src/server/set_family.cc

```diff
@@ -1344,6 +1344,56 @@ void SMembers(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void SRandMember(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 0);
+  int count = 1;
+  if (args.size() == 2) {
+    string_view arg = ArgS(args, 1);
+    if (!absl::SimpleAtoi(arg, &count)) {
+      (*cntx)->SendError(kInvalidIntErr);
+      return;
+    }
+  } else if (args.size() > 2) {
+    return (*cntx)->SendError(absl::StrCat("unsupported option ", ArgS(args, 2)));
+  }
+  const unsigned ucount = std::abs(count);
+
+  const auto cb = [&](Transaction* t, EngineShard* shard) -> OpResult<StringVec> {
+    StringVec result;
+    OpResult<PrimeIterator> find_res = shard->db_slice().Find(t->GetDbContext(), key, OBJ_SET);
+    if (!find_res) {
+      return find_res.status();
+    }
+
+    PrimeValue& pv = find_res.value()->second;
+    if (IsDenseEncoding(pv)) {
+      StringSet* ss = (StringSet*)pv.RObjPtr();
+      ss->set_time(MemberTimeSeconds(t->GetDbContext().time_now_ms));
+    }
+
+    container_utils::IterateSet(find_res.value()->second,
+                                [&result, ucount](container_utils::ContainerEntry ce) {
+                                  result.push_back(ce.ToString());
+                                  return result.size() != ucount;
+                                });
+    return result;
+  };
+
+  OpResult<StringVec> result = cntx->transaction->ScheduleSingleHopT(cb);
+
+  if (result || result.status() == OpStatus::KEY_NOTFOUND) {
```

Comment:
please note that the output format changes if count is present or not. if count is absent, it returns an element or nil,  otherwise it returns an array.